pkg/cloud/gcp/gcs_storage.go

```diff
@@ -343,6 +345,24 @@ func (g *gcsStorage) Close() error {
 	return g.client.Close()
 }
 
+func shouldRetry(err error) bool {
+	if defaultShouldRetry(err) {
+		return true
+	}
+
```

Comment:
maybe add a small comment on these specific cases are outside google's default retries, and why we retry them?

Reply:
done